lib/extensions/sign_in_with_callback/devise/controllers/sign_in_out.rb

```diff
@@ -0,0 +1,21 @@
+module Extensions::SignInWithCallback::Devise::Controllers::SignInOut
+  class << self
+    def included(base)
+      base.alias_method_chain :sign_in, :callback
+    end
+  end
+
+  def sign_in_with_callback(resource_or_scope, *args)
+    original_args = args.dup
+    original_args.extract_options!
+    resource = original_args.last || resource_or_scope
+
+    if resource.respond_to?(:_sign_in_callbacks)
```

Comment:
You can provide a module to include support for the sign_in callback. Then when signing in, check if that object has the module, if so, trigger the callback.